src/common/plugin/PluginState.ts

```diff
@@ -219,9 +221,31 @@ export function makePluginState(settings: PluginStateSettings): PluginState {
     },
 
     async updateServers(settings: UtxoUserSettings): Promise<void> {
-      enableCustomServers = settings.enableCustomServers
-      customServers = settings.blockbookServers
+      if (settings === userSettings) return
+
+      const hasServerListChanged = (() => {
```

Comment:
Can this be a normal function, and then just invoke it in the `if` statement as `if (!hasServerListChanged() && ...`?

IIFE's are always confusing to read, because they look a lot like function definitions when scanning the code. In this case, it seems like an actual function definition would have worked just as well.